### PR TITLE
Replace the HTTP mock methods with a transport adapter

### PR DESCRIPTION
**Pull request by tonybaloney**
### Description

This change is to the base test classes. At the moment they mock out the response, raw response and connection classes. Which is fine but it's led to issues that could have been uncovered in the unit tests because it doesn't represent how the drivers actually respond.

This change will replace the mock connection and mock response classes with the actual implementations and use a `requests_mock` session context manager to get the response from the existing methods so hopefully I won't need to change each of the test classes.

For more information on contributing, please see [Contributing](http://libcloud.readthedocs.org/en/latest/development.html#contributing)
section of our documentation.

### Status

Replace this: describe the PR status. Examples:

- work in progress
- done, ready for review

### Checklist (tick everything that applies)

- [x] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [ ] Documentation
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [x] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

**Comment by tonybaloney** [comment]
**Linode**: 
Tests fail because the in `secrets.py-dist` the tuple is setup for connectionUserAndKey, but Linode is key only.

File: libcloud/common/base.py [hunk 1]
```diff
@@ -236,6 +236,7 @@ def parse_body(self):
                 # lxml wants a bytes and tests are basically hard-coded to str
                 body = ET.XML(self.body.encode('utf-8'))
         except:
+            import pdb; pdb.set_trace()
```
Comment (pquentin) [inline libcloud/common/base.py]: note: debugger called (also, pudb is nice)

**Comment by tonybaloney** [comment]
uncovered an issue with backblaze b2:
https://github.com/apache/libcloud/blob/trunk/libcloud/storage/drivers/backblaze_b2.py#L313-L322 the API is wrong. it should take an iterator as the second field.

**Comment by tonybaloney** [comment]
uncovered that vsphere driver is totally broken

**Comment by tonybaloney** [comment]
the remaining issues are almost entirely to do with encoding of fixtures. Some (e.g. aliyun ecs) have different encodings in the same fixtures folder. e.g.
```
(python2) Anthonys-MacBook-Pro-2:ecs anthonyshaw$ file -I *.xml
attach_disk.xml:                        application/xml; charset=us-ascii
copy_image.xml:                         application/xml; charset=us-ascii
create_disk.xml:                        application/xml; charset=us-ascii
create_image.xml:                       application/xml; charset=us-ascii
create_instance.xml:                    application/xml; charset=us-ascii
create_node_describe_instances.xml:     application/xml; charset=us-ascii
create_public_ip.xml:                   application/xml; charset=us-ascii
create_security_group.xml:              application/xml; charset=us-ascii
create_snapshot.xml:                    application/xml; charset=us-ascii
create_volume_describe_disks.xml:       application/xml; charset=us-ascii
delete_disk.xml:                        application/xml; charset=us-ascii
delete_image.xml:                       application/xml; charset=us-ascii
delete_instance.xml:                    application/xml; charset=us-ascii
delete_security_group_by_id.xml:        application/xml; charset=us-ascii
delete_snapshot.xml:                    application/xml; charset=us-ascii
describe_disks.xml:                     application/xml; charset=us-ascii
describe_images.xml:                    inode/x-empty; charset=binary
describe_instance_types.xml:            application/xml; charset=us-ascii
describe_instances.xml:                 application/xml; charset=us-ascii
describe_regions.xml:                   application/xml; charset=utf-8
describe_security_group_attributes.xml: application/xml; charset=us-ascii
describe_security_groups.xml:           application/xml; charset=us-ascii
describe_snapshots.xml:                 application/xml; charset=us-ascii
describe_zones.xml:                     application/xml; charset=utf-8
destroy_node_describe_instances.xml:    application/xml; charset=us-ascii
destroy_volume_describe_disks.xml:      application/xml; charset=us-ascii
detach_disk.xml:                        application/xml; charset=us-ascii
detach_volume_describe_disks.xml:       application/xml; charset=us-ascii
get_image_describe_images.xml:          application/xml; charset=us-ascii
join_security_group_by_id.xml:          application/xml; charset=us-ascii
leave_security_group_by_id.xml:         application/xml; charset=us-ascii
pages_describe_images.xml:              application/xml; charset=utf-8
pages_describe_images_page2.xml:        application/xml; charset=utf-8
reboot_instance.xml:                    application/xml; charset=us-ascii
reboot_node_describe_instances.xml:     application/xml; charset=us-ascii
start_instance.xml:                     application/xml; charset=us-ascii
stop_instance.xml:                      application/xml; charset=us-ascii
stop_node_describe_instances.xml:       application/xml; charset=us-ascii
```

**Comment by tonybaloney** [comment]
outstanding bug in the azure service management driver, see:
https://github.com/apache/libcloud/pull/1031/commits/5732852ad92d582c136ab6f7c836b262f12051bc#diff-593dee57ae65e97680b29083f4c63bb8R389

**Comment by tonybaloney** [comment]
aliyun responses have been converted to ASCII. This is a bit of a hack, but since the File Fixtures module loads them into string objects, it's unavoidable without getting into a world of py2 v.s. py3 pain.

**Comment by tonybaloney** [comment]
this was actually a bug in the Aliyun ECS driver!

**Comment by tonybaloney** [comment]
Also reintroduced coveralls back into the build pipeline. This time I'm 10x more familiar with tox and how not to break it.

**Comment by tonybaloney** [comment]
this is done. found 3 bugs.

- Aliyun ECS, when responses contain UTF-8 encoding and user is Python 2.7, it fails to parse the response body (fixed)
- Azure Classic driver would never list VhdImages, also list public IPs would be broken
- vSphere driver would not authenticate since 2.0

Merging

**Pull request merged**